File: src/main/java/destiny/core/calendar/eightwords/onePalm/PositiveGenderImpl.java
/**
 * Created by smallufo on 2015-05-18.
 */
package destiny.core.calendar.eightwords.onePalm;

import destiny.core.Gender;
import destiny.core.chinese.EarthlyBranches;

import java.io.Serializable;
import java.util.Locale;

public class PositiveGenderImpl implements PositiveIF , Serializable {

  @Override
  public boolean isPositive(Gender gender, EarthlyBranches yearBranch) {
    return (gender == Gender.男);
  }

  @Override
  public String getTitle(Locale locale) {
    return "男順女逆";
  }

  @Override
  public String getDescription(Locale locale) {
    return "固定男順女逆";
  }
}

File: src/main/kotlin/destiny/core/chinese/onePalm/PositiveGenderImpl.kt
/**
 * Created by smallufo on 2015-05-18.
 */
package destiny.core.chinese.onePalm

import destiny.core.Descriptive
import destiny.core.Gender
import destiny.core.chinese.Branch
import destiny.tools.asDescriptive
import destiny.tools.JSerializable

class PositiveGenderImpl : IPositive,
                           Descriptive by PositiveImpl.Gender.asDescriptive(),
                           JSerializable {

  override fun isPositive(gender: Gender, yearBranch: Branch): Boolean {
    return gender === Gender.M
  }
}
